### PR TITLE
Authentication examples for Node.js and Python

### PR DESCRIPTION
**Pull request by bartolootrit**
There is a lack of examples with custom headers for authentication and disabled hostname checks.

### CODE REVIEW

**Comment by chalin** [comment]
Thanks for the contribution @bartolootrit! You'll need to agree to the CLA before we can consider this PR.

**Comment by chalin** [comment]
Hi all. I've added a bunch of folks as potential reviewers. If you don't think that you're the right person to review this PR, please add someone else who you believe might be better qualified.

Note that the current "generic"/"catch-all languages" Authentication page will likely be reworked into language-specific versions, and moved into each language's section (#543) -- as was done for the ALTS page (#527). But this probably won't happen until the docsy migration is complete (#479)

cc @ejona86

File: content/docs/guides/auth.md [hunk 1]
```diff
@@ -505,8 +579,17 @@ var stub = new helloworld.Greeter('localhost:50051', grpc.credentials.createInse
...
 
 ```js
-var ssl_creds = grpc.credentials.createSsl(root_certs);
-var stub = new helloworld.Greeter('myservice.example.com', ssl_creds);
+const root_cert = fs.readFileSync('path/to/root-cert');
+const ssl_creds = grpc.credentials.createSsl(root_cert);
+const stub = new helloworld.Greeter('myservice.example.com', ssl_creds);
+```
+
+##### With server authentication SSL/TLS and disabled hostname checking
```
Comment (murgatroid99) [inline content/docs/guides/auth.md:587]: I don't see "disabled hostname checking" examples for the other languages, and I don't think we should add it. That option is intended only for testing and I would want to at least make that clearer if we are going to recommend it here.

Comment (bartolootrit) [reply content/docs/guides/auth.md]: Done

**Review by murgatroid99 — APPROVED** [review]
The Node changes look good to me now.

File: content/docs/guides/auth.md [hunk 2]
```diff
@@ -421,6 +421,76 @@ channel = google_auth_transport_grpc.secure_authorized_channel(
...
+            context.abort(grpc.StatusCode.UNAUTHENTICATED, 'Invalid key')
+
+        self._deny = grpc.unary_unary_rpc_method_handler(deny)
+
+    def intercept_service(self, continuation, handler_call_details):
+        if (
+            handler_call_details.invocation_metadata
+            and handler_call_details.invocation_metadata[0] == self._valid_metadata
+        ):
+            return continuation(handler_call_details)
```
Comment (gnossen) [inline content/docs/guides/auth.md:476]: Nit: An explicit `else` would be nice here to aid readability, especially considering the indentation.

**Review by gnossen — APPROVED** [review]
Thanks for the contribution! The Python changes look good!

**Comment by chalin** [comment]
Thanks for the reviews all!

File: content/docs/guides/auth.md (hunk 2 above)
Comment (chalin) [reply content/docs/guides/auth.md:476]: @bartolootrit - can you make this update?

Comment (bartolootrit) [reply content/docs/guides/auth.md:476]: Done

Comment (chalin) [reply content/docs/guides/auth.md:476]: Thanks!

**Pull request merged**